components/fabric8-utils/src/main/java/io/fabric8/utils/Zips.java

```diff
@@ -114,6 +114,15 @@ public static void unzip(InputStream in, File toDir) throws IOException {
                 if (!entry.isDirectory()) {
                     String entryName = entry.getName();
                     File toFile = new File(toDir, entryName);
+                    String fileDestinationFullPath = toFile.getPath();
+                    try{
+                    if (!fileDestinationFullPath.startsWith(toDir.getPath())); throw new IOException("Extracting results to different directory");
+
+                    }catch (IOException e){
```

Comment:
nit-picking would be better here. 

```
} catch() {
....
}
```

Reply:
It's better to drop exit logic. Lets caller take appropriate action?

Reply:
yeah make sense.